### PR TITLE
Clarify "It is an error to..." wording for zero-duration behaviors.

### PR DESCRIPTION
**Pull request by frewsxcv**
Documentation fix side of https://github.com/rust-lang/rust/issues/48311.

### CODE REVIEW

File: src/libstd/net/tcp.rs [hunk 1]
```diff
@@ -282,6 +284,20 @@ impl TcpStream {
     ///                        .expect("Couldn't connect to the server...");
     /// stream.set_read_timeout(None).expect("set_read_timeout call failed");
     /// ```
+    ///
+    /// An [`Err`] is returned if the zero [`Duration`] is passed to this
```
Comment (sfackler) [inline src/libstd/net/tcp.rs]: I don't know if we need to have an explicit example of an invalid call.

Comment (steveklabnik) [reply src/libstd/net/tcp.rs]: We talked about it at the docs team meeting, and everyone was generally in favor of showing off error conditions.

**Comment by frewsxcv** [comment]
tests are passing now, let me know if anyone sees anything else!

**Comment by shepmaster** [comment]
Heads-up @sfackler, this is ready for re-review.

**Pull request merged**